generate-new/vk/VK/SPIRVElements.hs

```diff
@@ -0,0 +1,280 @@
+module VK.SPIRVElements
+  ( renderSPIRVElements
+  ) where
+
+import           CType                          ( CType(TypeName) )
+import           Data.Bits
+import           Data.Foldable
+import           Data.List.Extra                ( nubOrd )
+import qualified Data.Text                     as T
+import           Data.Text.Prettyprint.Doc
+import           Data.Vector                    ( Vector )
+import qualified Data.Vector                   as V
+import           Data.Version                   ( Version(versionBranch)
+                                                , makeVersion
+                                                )
+import           Error
+import           Haskell                        ( HName(..)
+                                                , allTypeNames
+                                                , renderType
+                                                , (~>)
+                                                )
+import           Language.Haskell.TH            ( Type(ConT)
+                                                , mkName
+                                                )
+import           Polysemy.Input
+import           Relude
+import           Render.Element
+import           Render.SpecInfo                ( HasSpecInfo
+                                                , SpecInfo(..)
+                                                )
+import           Render.Type                    ( cToHsType )
+import           Render.Type.Preserve           ( Preserve(DoNotPreserve) )
+import           Spec.Types
+import           Text.InterpolatedString.Perl6.Unindented
+import           VkModulePrefix
+import qualified Prelude
+
+renderSPIRVElements
+  :: (HasErr r, HasRenderParams r, HasSpecInfo r)
+  => Vector SPIRVExtension
+  -> Vector SPIRVCapability
+  -> Sem r RenderElement
+renderSPIRVElements exts caps = genRe "SPIR-V stuff" $ do
+  tellExplicitModule (vulkanModule ["SPIRVRequirements"])
+  tellCanFormat
+  bespokeStuff
+  renderExts exts
+  renderCaps caps
+
+renderExts
+  :: (HasRenderElem r, HasRenderParams r, HasErr r, HasSpecInfo r)
+  => Vector SPIRVExtension
+  -> Sem r ()
+renderExts = renderSPIRVThing "spirvExtensionRequirements"
+                              spirvExtensionName
+                              spirvExtensionReqs
+
+renderCaps
+  :: (HasRenderElem r, HasRenderParams r, HasErr r, HasSpecInfo r)
+  => Vector SPIRVCapability
+  -> Sem r ()
+renderCaps = renderSPIRVThing "spirvCapabilityRequirements"
+                              spirvCapabilityName
+                              spirvCapabilityReqs
+
+renderSPIRVThing
+  :: (HasRenderElem r, HasRenderParams r, HasErr r, HasSpecInfo r)
+  => Text
+  -> (a -> Text)
+  -> (a -> Vector SPIRVRequirement)
+  -> Vector a
+  -> Sem r ()
+renderSPIRVThing funName name reqs xs = do
+  let case' x = do
+        let mergeReqs = Prelude.head -- TODO: this is probably wrong! discarding the
+                                     -- other reqs
```

Comment:
This is not correct! It's only paying attention to the first "enable" node for each capability/requirement

See here for an example: https://github.com/KhronosGroup/Vulkan-Docs/blob/9f9fd160c8dbd8752447a4a8a2e32fbb92b9722d/xml/vk.xml#L14907-L14910

Reply:
I think the best thing to do would be to also take a version, and if it is at least a version mentioned in the "enable" then that enable is returned, otherwise the other one is.